# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/BaselineJIT.cpp
# Commit: 49a2da59aa3e
# Full Hash: 49a2da59aa3e33b3719c7953a98fac6c24b8a514
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2019-07-21 21:59:35
# Regressor Bug: 1566330
# File Overlap Count: 2
# Description:
#   Bug 1566330 - Let BaselineDebugModeOSR resume in the interpreter, remove BaselineDebugModeOSRInfo. r=iain
#   
#   At this point most of the DebugModeOSR complexity came from dealing with the
#   On -> Off case because debugger callVMs are not present in the recompiled script.
#   We also had to worry about loading unsynced stack values in R0/R1 in the
# ==============================================================================

diff -r ad5e0980c273 -r 49a2da59aa3e js/src/jit/BaselineJIT.cpp
--- a/js/src/jit/BaselineJIT.cpp	Sun Jul 21 09:27:47 2019 +0000
+++ b/js/src/jit/BaselineJIT.cpp	Fri Jul 19 09:01:45 2019 +0000
@@ -440,8 +440,7 @@
 
 BaselineScript* BaselineScript::New(
     JSScript* jsscript, uint32_t bailoutPrologueOffset,
-    uint32_t warmUpCheckPrologueOffset, uint32_t debugOsrPrologueOffset,
-    uint32_t debugOsrEpilogueOffset, uint32_t profilerEnterToggleOffset,
+    uint32_t warmUpCheckPrologueOffset, uint32_t profilerEnterToggleOffset,
     uint32_t profilerExitToggleOffset, size_t retAddrEntries,
     size_t pcMappingIndexEntries, size_t pcMappingSize, size_t resumeEntries,
     size_t traceLoggerToggleOffsetEntries) {
@@ -473,7 +472,6 @@
   }
   new (script)
       BaselineScript(bailoutPrologueOffset, warmUpCheckPrologueOffset,
-                     debugOsrPrologueOffset, debugOsrEpilogueOffset,
                      profilerEnterToggleOffset, profilerExitToggleOffset);
 
   size_t offsetCursor = sizeof(BaselineScript);
@@ -1120,7 +1118,8 @@
                                uint32_t profilerExitToggleOffset,
                                CodeOffsetVector&& debugInstrumentationOffsets,
                                CodeOffsetVector&& debugTrapOffsets,
-                               CodeOffsetVector&& codeCoverageOffsets) {
+                               CodeOffsetVector&& codeCoverageOffsets,
+                               const CallVMOffsets& callVMOffsets) {
   code_ = code;
   interpretOpOffset_ = interpretOpOffset;
   interpretOpNoDebugTrapOffset_ = interpretOpNoDebugTrapOffset;
@@ -1129,6 +1128,7 @@
   debugInstrumentationOffsets_ = std::move(debugInstrumentationOffsets);
   debugTrapOffsets_ = std::move(debugTrapOffsets);
   codeCoverageOffsets_ = std::move(codeCoverageOffsets);
+  callVMOffsets_ = callVMOffsets;
 }
 
 bool jit::GenerateBaselineInterpreter(JSContext* cx,